### PR TITLE
#18 and #19 Fix to include filters for designs and extention to create AEM 6.2 Editable Templates folders structure

### PR DESCRIPTION
**Pull request by mtstv**
Fixed, please merge.

### CODE REVIEW

**Comment by mtstv** [comment]
I reverted the last 2 commits in the readme due to closing the issue #21 .
Please merge.

File: templates/aem-multimodule-project/content/src/main/content/META-INF/vault/filter.xml [hunk 1]
```diff
@@ -4,6 +4,9 @@
 <% if (includeAcsAemCommons && enableErrorHandler) { %>
     <filter root="/apps/sling/servlet/errorhandler"/>
 <% } %>
+<% if (createEditableTemplatesStructure) { %>
+    <filter root="/conf/${appsFolderName}"/>
```
Comment (justinedelson) [inline templates/aem-multimodule-project/content/src/main/content/META-INF/vault/filter.xml]: IIUC, this is incorrect. With this in place, installation of the package will overwrite any existing templates, which really makes them non-editable (sure, you can edit them, but the next deployment will blow away your edits). TBH, I'm not 100% sure what the right answer is to this.

One choice would be to exclude the "editable" directories from the package. Another would be to use merge.

File: templates/aem-multimodule-project/lazybones.groovy [hunk 2]
```diff
@@ -335,6 +341,100 @@ def installDir = new File(projectDir, "content/src/main/content/jcr_root/apps/${
...
+	def confDir = new File(projectDir, "content/src/main/content/jcr_root/conf")
+	confDir.mkdirs()
+    writeToFile(confDir, ".content.xml", """\
+<?xml version="1.0" encoding="UTF-8"?>
+<jcr:root xmlns:sling="http://sling.apache.org/jcr/sling/1.0" xmlns:jcr="http://www.jcp.org/jcr/1.0" xmlns:rep="internal"
+    jcr:mixinTypes="[rep:AccessControllable]"
+    jcr:primaryType="sling:Folder"/>
+""")
+
+	def confProjectDir = new File(confDir, "${props.appsFolderName}")
```
Comment (justinedelson) [inline templates/aem-multimodule-project/lazybones.groovy]: WDYT about making this folder name configurable?

File: templates/aem-multimodule-project/content/src/main/content/META-INF/vault/filter.xml (hunk 1 above)
Comment (mtstv) [inline templates/aem-multimodule-project/content/src/main/content/META-INF/vault/filter.xml]: I think merge will be Ok. Just need to test it.

File: templates/aem-multimodule-project/lazybones.groovy [hunk 3]
```diff
@@ -182,6 +182,12 @@ def defaultFolderName = transformText(props.projectName, from: NameType.NATURAL,
 props.appsFolderName = ask("Folder name under /apps for components and templates [${defaultFolderName}]: ", defaultFolderName, "appsFolderName")
 props.contentFolderName = ask("Folder name under /content which will contain your site [${defaultFolderName}] (Don't worry, you can always add more, this is just for some default configuration.): ", defaultFolderName, "contentFolderName")
 
+// Create AEM 6.2 Editable Templates folders? 
+props.createEditableTemplatesStructure = ''
+if (props.aemVersion == VERSION_62) {
+	props.createEditableTemplatesStructure = askBoolean("Would you like to create AEM 6.2 Editable Templates folders? [yes]: ", "yes", );
```
Comment (mtstv) [inline templates/aem-multimodule-project/lazybones.groovy]: I copied the logic from here and thought it was right way, isn't it?

```
props.enableClassicAuthoringAsDefault = ''
if (props.aemVersion == VERSION_60) {
    props.enableClassicAuthoringAsDefault = askBoolean("Do you want to set the default authoring UI to Classic UI? [yes]: ", "yes", )
}

```

File: templates/aem-multimodule-project/lazybones.groovy (hunk 2 above)
Comment (mtstv) [inline templates/aem-multimodule-project/lazybones.groovy]: we can do it, but I think the value will be the same with ${props.appsFolderName}"

File: templates/aem-multimodule-project/lazybones.groovy (hunk 3 above)
Comment (mtstv) [inline templates/aem-multimodule-project/lazybones.groovy]: I used groovy first time in my life, just copied the existing code.
:-)

**Comment by godanny86** [comment]
coming to the party a little late... WDYT of creating a separate content package ui.content for artifacts like editable templates under /conf but also maybe a base /content structure. Filters would default to merge. Would be more inline with the We.Retail reference implementation...

**Comment by justinedelson** [comment]
@godanny86 I think that is an OK suggestion, except that `/conf/<appname>/settings/wcm/template-types` needs to be in `ui.apps`, not `ui.content` since `template-types` is under developer control.

**Comment by godanny86** [comment]
good point @justinedelson that template-types would be under developer control. What about a filter.xml that defaults to:
`<filter root='/conf/<appname>/settings/wcm/template-types' />`
`<filter root='/conf/<appname>/settings/wcm/templates' mode='merge' />`
`<filter root='/conf/<appname>/settings/wcm/policies' mode='merge' />`

I feel like there will be other filters under /conf like metadata-schemas, customized search facets that fall into a 'gray' area that can be updated via the UI by power users but also many projects may wish to manage under source control. Maybe **ui.content** isn't the correct name, could do **ui.conf** ?

**Comment by justinedelson** [comment]
@godanny86 I think just the merge on the top-level will suffice. I think @mtstv already said he was going to add this.

**Comment by godanny86** [comment]
cool, that works

File: templates/aem-multimodule-project/lazybones.groovy (hunk 3 above)
Comment (justinedelson) [inline templates/aem-multimodule-project/lazybones.groovy]: I've fixed this typo in 8e27bd69f60cff7b45e6a10f908f7dc253e759d7

Comment (mtstv) [inline templates/aem-multimodule-project/lazybones.groovy]: I fixed my code also

File: templates/aem-multimodule-project/lazybones.groovy (hunk 2 above)
Comment (mtstv) [inline templates/aem-multimodule-project/lazybones.groovy]: I made the /conf/<confFolderName> folder configurable.

**Comment by mtstv** [comment]
Fixed issues form review.

**Comment by mtstv** [comment]
One thing more, I think the file is unnecessary and can be deleted, because it will be not used by content-package-maven-plugin during the package generation.
/example-project.ui.apps/src/main/content/META-INF/vault/definition/.content.xml

**Comment by godanny86** [comment]
@justinedelson so can @mtstv updates be merged?

**Comment by justinedelson** [comment]
yeah, I'll try to get to that this week

**Comment by godanny86** [comment]
sweet!

**Pull request merged**